integration/tests/params.go

```diff
@@ -217,3 +226,39 @@ func NewParams(clusterNamePrefix string) *Params {
 	params.GenerateCommands()
 	return &params
 }
+
+// attemptSettingUserID will attempt to fetch the first 2 characters of the userID running the
+// integration tests to be used as a prefix for cluster names. It fails silently if an error occurs
+func (p *Params) attemptSettingUserID() {
+	clusterProvider, err := eks.New(&api.ProviderConfig{Region: p.Region}, &api.ClusterConfig{
+		TypeMeta: api.ClusterConfigTypeMeta(),
+		Metadata: &api.ClusterMeta{
+			Version: p.Version,
+			Name:    p.ClusterName,
+			Region:  p.Region,
+		},
+	})
+
+	msg := "Warning: failed to get UserID, proceeding without configuring cluster name prefix. Error: %w"
+	if err != nil {
+		fmt.Fprintf(GinkgoWriter, msg, err)
+		return
+	}
+
+	out, err := clusterProvider.Provider.STS().GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		fmt.Fprintf(GinkgoWriter, msg, err)
+		return
+	}
+
+	//example value: ABCD1234:my-name@email-provider
+	parts := strings.Split(*out.UserId, ":")
```

Comment:
it isn't required, see the func description :smile: 

Its worked just basing it off my local testing. We don't need to support all the different scenarios, at the end of the day this is for to make our lives easier. If the value is like this for me it should be the same for the rest of the team :smile: 

feel free to give it a run!

Reply:
My concern was specifically about the integration test account/CI, and I don't have access to it so I can't try it out!

Also, from the go docs about `strings.Split()`: `If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.`

Does this mean that it could log the first 2 chars of the test account? 🤔

Reply:
> Does this mean that it could log the first 2 chars of the test account? thinking Which I guess is not that bad anyway since the CI repo is private but hm

I think it contained some sensitive looking stuff, so I wasn't in favour of this but it is an option

> My concern was specifically about the integration test account/CI, and I don't have access to it so I can't try it out!

I'm less worried about this, either it works which is fine or it doesn't, which doesn't matter since this is mainly being added for us to differentiate our runs in our account

Reply:
I'm also not in favour of that, I'm saying that that is what this PR does: it prints the first 2 chars of the `userID`, and in the case where the integration tests are run by the integration test account from the CI repo, this might be the AWS Account ID. 

I could be wrong and it doesn't print those 2 chars, and also maybe we don't care about printing those 2 first chars in a private repo (but it's still sensitive info)

Reply:
ahh I get what your saying. I think that won't happen in this scenario, it only works if the string matches the format "abc:name@email", and we don't have any email associated that contains the account id (or any sensitive information).

I'll give it a run though to verify :smile:

Reply:
yep, since the integration account doesn't use an email its blank `"eksctl-it-addons-adorable-mushroom-1645710242-cluster"`

Reply:
thank you for checking :D